src/content/en/updates/posts/2016/07/infinite-scroller.md

```diff
@@ -0,0 +1,246 @@
+
+<img src="/web/updates/images/2016/07/infinite-scroller/anchoring.png">
+
+After things in the runway have changed, we can now restore a situation that feels almost identical to the user. Win! Except a resized window means that each items has potentially changed its height, so how do we know how far down the anchored content should be placed? We don’t! To find out we would have to layout every element above the anchored item and add up all of their heights; this could cause a significant pause after a resize, and we don’t want that. Instead, we resort to assuming that every item above is the same size as a tombstone and adjust our scroll position accordingly. As elements are scrolled into the runway, we adjust or scroll position, effectively deferring the layout work to when it is actually needed.
+
+#Layout
+I have skipped over an important detail: Layout. Each recycling of a DOM element would normally relayout the entire runway which would bring us well below our target of 60 frames per second. To avoid this, we are taking the burden of layout onto ourselves and use absolutely positioned elements with transforms. This way we can pretend that all the elements further up the runway are still taking up space when in actuality there is only empty space. Since we are doing layout ourselves, we can cache the positions each item ends up and we can immediately load the correct element from cache when the user scrolls backwards.
+
+Ideally, items would only get repainted once when they get attached to the DOM and be unfazed by additions or removals of other items in the runway. That is possible, but only with modern browsers.
+
+#Bleeding-edge tweaks
+Recently, [Chrome added support for CSS Containment][Containment], a feature that allows us developers to tell the browser an element is a boundary for layout and paint work. Since we are doing layout ourselves here, it’s a prime application for containment. Whenever we add an element to the runway, we *know* the other items don’t need to be affected by the relayout. So each item should be get `contain: layout`. We also don‘t want to affect the rest of our website, so the runway itself should get this style directive as well.
+
+Another thing we considered is using [IntersectionObservers] as a mechanism to detect when the user has scrolled far enough for us to start recycling elements and load new data. However, IntersectionObservers are specified to be high latency (as if using `requestIdleCallback`), so we might actually *feel* less responsive with IntersectionObservers than without. Even our current implementation using the `scroll` event suffers from this problem, as scroll events are dispatched on a “best effort” basis. Eventually, [Houdini’s Compositor Worklet][Compositor Worklet] would be the high fidelity solution to this problem.
```

Comment:
I think part of the link is missing. Also IntersectionObserver should have backticks.